### ISSUE TITLE
Reaver t6x Port

### ISSUE THREAD

**Issue by hitokirironinzero** [chrisk44/Hijacker#11]
I know this isn't an actual issue and I don't know if it interests you for your project or not, but I've managed to compile the current version of Reaver t6x to android in a static pie object for my nexus 5. I've also added the macchanger attack from https://github.com/gabrielrcouto/reaver-wps. Its all setup to work with the NDK and one of the commits is a stand alone for easy compilation to play around with. Everything is on my github if you want to check it out.

**Comment by hitokirironinzero** [comment]
https://github.com/hitokirironinzero/android-reaver-t6x

**Comment by chrisk44** [comment]
You mean the version with the pixie-dust attack?

**Comment by hitokirironinzero** [comment]
Yeah that's the one https://github.com/t6x/reaver-wps-fork-t6x. Everything compiles fine and the Reaver binary works good and hasnt given me a issue yet. Wash compiles but is giving me a few issues needs a little tweaking

**Comment by chrisk44** [comment]
Ok, I'll take a look when I have the time and probably use it so the app won't need the nethunter chroot environment.

**Comment by chrisk44** [comment]
Could you upload a binary for arm? I don't have a linux environment set up right now.

**Comment by hitokirironinzero** [comment]
I've added two precompiled binaries in my repo, same ones I'm using on my phone. Check the bin folder I named the with -t6x suffix

**Comment by hitokirironinzero** [comment]
I also put a precompiled pixiewps binary with the other two incase you need that as well.

**Comment by hitokirironinzero** [comment]
I just cleaned up and improved the port everything should all work better now. I updated the binaries too. The default conf directory is configured to the /sdcard/ for now

**Comment by chrisk44** [comment]
The binary you have uploaded doesn't associate with a network but starts trying pins. It says:
[!] WARNING: Failed to assiciate with <mac> (ESSID: <essid>)
constantly...

The version I currently have in my app manages to associate with a network but then it always says "receive timeout occured".

I'm not entirely certain the the version I have actually associates with the network. My router is WPS-enabled but I have never successfully attacked it using reaver, not with my app, not even manually with my laptop.

**Comment by hitokirironinzero** [comment]
Alright that's good to know, I'll play around with the code and both binaries a bit and I think I might build the old repo of android-reaver-wps while I'm at it, see what I can workout.

**Comment by chrisk44** [comment]
Is there any progress with it? I would like to include it in the app if it's solid and discard the nethunter solution.

**Comment by hitokirironinzero** [comment]
Sorry mate not much progress at the moment, I started a new job and haven't had time to work on anything lately. I haven't abandoned the project and I've got a bit of time to work on it coming up, will keep you posted.

**Comment by 0v3rfl00w** [comment]
Anyone still working on this ? If it works this would be great step for the hijacker application

**Comment by Meterpretator** [comment]
You know, someone have been working on pixiewps on android. You may check if you hadn't https://github.com/aanarchyy/pixiewps-android

**Comment by chrisk44** [comment]
This is just the pixie dust attack. What I need is the reaver cracking tool that includes the attack and executes it on its own. Like the one linked above.

**Comment by 900n900** [comment]
You may check if you hadn't ""http://wpspin.blogspot.com "".I downloaded from there reaver 1.5.2 & bully with pixiewps attack for my S2 but i don't know if they are compilated for  android marshmellow so i don't know if they work in nexus 5.sorry for my pour english and i hope this may help you

**Comment by 0v3rfl00w** [comment]
@900n900 
So this is all about adfly money . what a waste of time

**Comment by EvilAirborn** [comment]
Can I get that reaver binary? I can't download from that website
